File: controllers/prescription_controller.go
package controllers

import (
	"encoding/json"
	"net/http"
	"time"

	"github.com/google/uuid"
)

type CreatePrescriptionRequest struct {
	PatientID string   `json:"patient_id" binding:"required"`
	Diagnosis string   `json:"diagnosis" binding:"required"`
	Medicines []string `json:"medicines" binding:"required"`
}

type PrescriptionController struct{}

func NewPrescriptionController() *PrescriptionController {
	return &PrescriptionController{}
}

func (c *PrescriptionController) CreatePrescription(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	var req CreatePrescriptionRequest
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusBadRequest)
		_ = json.NewEncoder(w).Encode(map[string]string{"error": "Invalid prescription payload"})
		return
	}

	prescriptionID := uuid.New()
	response := map[string]interface{}{
		"prescription_id": prescriptionID,
		"status":          "ISSUED",
		"patient_id":      req.PatientID,
		"diagnosis":       req.Diagnosis,
		"medicines":       req.Medicines,
		"digital_signature": map[string]string{
			"signed_by": r.Header.Get("X-Doctor-Public-ID"),
			"timestamp": time.Now().Format(time.RFC3339),
			"algorithm": "RSA-SHA256-PKCS1v15",
		},
		"message": "Digital prescription created successfully.",
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusCreated)
	_ = json.NewEncoder(w).Encode(response)
}

File: python_backend/controllers/prescription_controller.py
from fastapi import APIRouter, Header, HTTPException, status
from pydantic import BaseModel
from typing import List, Optional
import uuid
import datetime

router = APIRouter(prefix="/api/v1/prescriptions", tags=["prescriptions"])

class CreatePrescriptionRequest(BaseModel):
    patient_id: str
    diagnosis: str
    medicines: List[str]

@router.post("", status_code=status.HTTP_201_CREATED)
async def create_prescription(
    req: CreatePrescriptionRequest,
    x_doctor_public_id: Optional[str] = Header(None, alias="X-Doctor-Public-ID")
):
    prescription_id = str(uuid.uuid4())
    response = {
        "prescription_id": prescription_id,
        "status": "ISSUED",
        "patient_id": req.patient_id,
        "diagnosis": req.diagnosis,
        "medicines": req.medicines,
        "digital_signature": {
            "signed_by": x_doctor_public_id or "",
            "timestamp": datetime.datetime.now(datetime.timezone.utc).isoformat(),
            "algorithm": "RSA-SHA256-PKCS1v15"
        },
        "message": "Digital prescription created successfully."
    }
    return response
